contentprovider.go

```diff
@@ -72,34 +75,52 @@ func (p *contentProvider) newlines() []uint32 {
 	return p._nl
 }
 
-func (p *contentProvider) data(fileName bool) []byte {
-	if fileName {
+func (p *contentProvider) data(scope query.SearchScope) []byte {
+	switch scope {
+	case query.ScopeFileName:
 		return p.id.fileNameContent[p.id.fileNameIndex[p.idx]:p.id.fileNameIndex[p.idx+1]]
-	}
+	case query.ScopeFileContent:
+		if p._data == nil {
+			p._data, p.err = p.id.readContents(p.idx)
+			p.stats.FilesLoaded++
+			p.stats.ContentBytesLoaded += int64(len(p._data))
+		}
 
-	if p._data == nil {
-		p._data, p.err = p.id.readContents(p.idx)
-		p.stats.FilesLoaded++
-		p.stats.ContentBytesLoaded += int64(len(p._data))
+		return p._data
+	case query.ScopeSymbol:
+		return p.id.symbolNames[p.id.symbolIndex[p.idx]:p.id.symbolIndex[p.idx+1]]
```

Comment:
is it possible that p.idx isn't in the map. eg a plain text file with no symbols.